### PR TITLE
[FIX] tool-install: Support --ignore-failed-sources flag correctly

### PR DESCRIPTION
**Pull request by edvilme**
Description
dotnet tool install or update did not respect the --ignore-failed-sources flag. A fix for this was contributed on main but an automatic backport could not be made.

Testing
Additional unit tests to verify the changes were made, making sure the flag is being parsed accurately and that command execution matches desired results. Manual testing was also made.

Risk
Low - implements missing logic for flag. 

Regression
No

Customer Impact
Allows customers to install or update tools even if one or more of their NuGet feeds is invalid. 

Backport of #44309 to release/8.0.1xx

### CODE REVIEW

**Review by nagilson — APPROVED** [review]
The change looks good to me :) 

Don't merge this yet, though! We need to get servicing approval.
Follow the template I filled out in https://github.com/dotnet/sdk/pull/44203 with a description, testing, etc. Manual testing of the build on this branch to confirm the fix is required. It's helpful if you provide a screenshot of evidence that it works.

The rest of the procedure is described in https://github.com/dotnet/sdk/pull/44309.

Even when you get servicing approval, you need to wait for the branch to be open. @marcpopMSFT Is the most aware of the schedule here of when you can next merge this PR. And unfortunately, I think youre going to need to backport to 9.0.1xx as well. Normally Id say mid November but the schedule is weird and not posted yet due to December being a less busy month.

**Comment by marcpopMSFT** [comment]
branding is usually done around the first Tuesday of each month. More on the process documented here: https://github.com/dotnet/sdk/blob/main/documentation/project-docs/SDK-PR-guide.md#servicing-releases Changes can be checked in after branding but as nagilson said, require approval.

**Comment by edvilme** [comment]
> And unfortunately, I think youre going to need to backport to 9.0.1xx as well

Can I try backporting with github bot on the original PR for that?

**Comment by nagilson** [comment]
It's worth trying!

**Pull request merged**
